doc/ci.jinja

```diff
@@ -0,0 +1,66 @@
+.. _continuous integration:
+
+Continuous integration
+======================
+
+Please note that if one of the following markers appear in the latest commit message, the following actions are taken.
+
+============================ ===================
+Commit message marker        Action taken by CI
+============================ ===================
+[skip ci]                    Gtihub CI is skipped completely. Several other options are also possible, see `github documentation <https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/skipping-workflow-runs>`_).
+[skip test]                  Skip running the tests.
+[skip doc]                   Skip building the doc.
+[test nightly]               Run tests on the nightly build of Nilearn's dependencies.
+[full doc]                   Run a full build of the documentation (meaning that all examples will be run).
+[example] name_of_example.py Run partial documentation build but will run the requested example.
+[force download]             Force a download of all the dataset required for the build of the documentation.
+============================ ===================
+
+Documentation build and deploy
+------------------------------
+
+.. using a mermaid flowchart: https://mermaid.js.org/syntax/flowchart.html
+
+.. mermaid:: mermaid/doc_deploy.mmd
+
+The development docs are built using GitHub Actions and Circle-CI.
+
+The workflow ``.github/workflows/build-docs.yml`` builds the doc and stores as a github artifact.
+On the ``main`` branch (or respectively on GitHub release),
+this workflow will also deploy the doc
+to https://github.com/nilearn//nilearn.github.io.git on the ``dev`` (respectively ``stable``).
+
+On pull-request,
+a successful build of the doc by ``.github/workflows/build-docs.yml`` triggers ``trigger_hosting_on_pr.yml``.
```

Comment:
Maybe we could still trigger the host_docs workflow regardless of linkcheck workflow status? WDYT, @Remi-Gau ?

I found that the linkcheck workflow sometimes just fails because of some temporarily unavailable webpages. The only way to make it pass is to monitor the failing link and re-run the linkcheck once the link is up.

Reply:
linkcheck is only run on main so it should not affect the deploy on PR, no?

Reply:
I think it is run when we request a `[full doc]` (had to deal with it several times on #5071)

Reply:
you are correct

Reply:
let's make sure it only runs on main

Reply:
easiest way is to extract into its own workflow I think: let's to it after this PR